Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/HostApp/ArcGISColorManager.cs

```diff
@@ -0,0 +1,352 @@
+    if (rowValue is int || rowValue is Int16 || rowValue is Int64)
+    {
+      newRowValue = newRowValue.Split(".")[0];
+      newGroupValue = newGroupValue.Split(".")[0];
+    }
+    else if (rowValue is double || rowValue is float)
+    {
+      newRowValue = string.Concat(
+        newRowValue.Split(".")[0],
+        ".",
+        newRowValue.Split(".")[^1].AsSpan(0, Math.Min(5, newRowValue.Split(".")[^1].Length))
+      );
+      newGroupValue = string.Concat(
+        newGroupValue.Split(".")[0],
+        ".",
+        newGroupValue.Split(".")[^1].AsSpan(0, Math.Min(5, newGroupValue.Split(".")[^1].Length))
+      );
+    }
+
+    return (newRowValue, newGroupValue);
+  }
+
+  private bool TryGetGraduatedRendererColor(
+    CIMClassBreaksRenderer graduatedRenderer,
+    Row row,
+    Dictionary<string, FieldDescription> fields,
+    out int color
+  )
+  {
+    if (!TryGetSymbolColor(graduatedRenderer.DefaultSymbol.Symbol, out color)) // get default color
+    {
+      return false;
+    }
+
+    string? usedField = null;
+    if (fields.TryGetValue(graduatedRenderer.Field, out FieldDescription? field))
+    {
+      usedField = field.Name;
+    }
+
+    List<CIMClassBreak> reversedBreaks = new(graduatedRenderer.Breaks);
+    reversedBreaks.Reverse();
+    foreach (var rBreak in reversedBreaks)
+    {
+      // keep looping until the last matching condition
+      if (Convert.ToDouble(row[usedField]) <= rBreak.UpperBound)
+      {
+        if (!TryGetSymbolColor(rBreak.Symbol.Symbol, out color)) // get default color
+        {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  }
+
+  // Tries to retrieve the feature layer color by renderer and row, or a default color of -1
+  private int GetLayerColorByRendererAndRow(CIMRenderer renderer, Row row, Dictionary<string, FieldDescription> fields)
+  {
+    // default color to white. this will be used if the renderer is not supported.
+    int color = -1;
+
+    // get color depending on renderer type
+    switch (renderer)
+    {
+      case CIMSimpleRenderer simpleRenderer:
+        if (!TryGetSymbolColor(simpleRenderer.Symbol.Symbol, out color))
+        {
+          // TODO: report partial success on color?
+        }
+        break;
+
+      // unique renderers have groups of conditions that may affect the color of a feature
+      // resulting in a different color than the default renderer symbol color
+      case CIMUniqueValueRenderer uniqueRenderer:
+        if (!TryGetUniqueRendererColor(uniqueRenderer, row, fields, out color)) // get default color
+        {
+          // TODO: report partial success on color, could not retrieve group color?
+        }
+        break;
+
+      case CIMClassBreaksRenderer graduatedRenderer:
+        if (!TryGetGraduatedRendererColor(graduatedRenderer, row, fields, out color)) // get default color
+        {
+          // TODO: report partial success on color, could not retrieve group color?
```

Comment:
Can we add `CONVERTED WITH WARNING` text here and other related parts to help ourselves later when we have